### PR TITLE
Cherry pick changes for 2.8.5 release

### PR DESCRIPTION
**Pull request by AndyZhang0707**
# rebase and merge

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

KAG-4786

Fix #_[issue number]_

### CODE REVIEW

File: build/openresty/patches/LuaJIT-2.1-20210510_01-ffi-arm64-macos-fix-vararg-call-handling.patch [hunk 1]
```diff
@@ -0,0 +1,62 @@
+From 521b367567dc5d91d7f9ae29c257998953e24e53 Mon Sep 17 00:00:00 2001
```
Comment (guanlan) [inline build/openresty/patches/LuaJIT-2.1-20210510_01-ffi-arm64-macos-fix-vararg-call-handling.patch:1]: Seems like we accidentally introduced those patches

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,12 @@
-FROM kong/kong:2.7.0
```
Comment (curiositycasualty) [inline .devcontainer/Dockerfile:1]: I doubt we want this.

File: .requirements [hunk 3]
```diff
@@ -10,3 +10,10 @@ LIBYAML_VERSION=0.2.5
 KONG_GO_PLUGINSERVER_VERSION=v0.6.1
 KONG_BUILD_TOOLS_VERSION=4.40.1
 KONG_NGINX_MODULE_BRANCH=0.2.1
```
Comment (curiositycasualty) [inline .requirements:12]: It seems pre-bazel, on the `2.8.4` tag, we were:
```
KONG_PACKAGE_NAME=kong
KONG_CONFLICTS=kong-enterprise-edition
KONG_LICENSE="ASL 2.0"

RESTY_VERSION=1.19.9.1
RESTY_LUAROCKS_VERSION=3.8.0
RESTY_OPENSSL_VERSION=1.1.1o
RESTY_PCRE_VERSION=8.45
LIBYAML_VERSION=0.2.5
KONG_GO_PLUGINSERVER_VERSION=v0.6.1
KONG_BUILD_TOOLS_VERSION=4.40.1
KONG_NGINX_MODULE_BRANCH=0.2.1
```

~ https://github.com/Kong/kong/blob/2.8.4/.requirements

Comment (curiositycasualty) [reply .requirements:12]: Unless we can't avoid it, if possible, we probably don't want to change any of these versions.

File: build/openresty/patches/LuaJIT-2.1-20210510_01-ffi-arm64-macos-fix-vararg-call-handling.patch (hunk 1 above)
Comment (Water-Melon) [reply build/openresty/patches/LuaJIT-2.1-20210510_01-ffi-arm64-macos-fix-vararg-call-handling.patch:1]: Yes, besides that, there are two patches related to M1 that I think can also be removed.

build/openresty/patches/lua-resty-core-0.1.22_03-make-resty.core.shdict-compatible-with-m1.patch
build/openresty/patches/lua-resty-core-0.1.22_04-make-resty.core.response-compatible-with-m1.patch

File: .requirements (hunk 3 above)
Comment (Water-Melon) [reply .requirements:12]: https://github.com/Kong/lua-kong-nginx-module/commit/f52a34c17af4543245ff79e3227b9d514b8cfa5c
https://github.com/Kong/kong/commit/edf183546ca23dc4b4e771b7325b0f483286ad09

It seems that some new features from version 0.2.2 of lua-kong-nginx-module are being used.

File: build/openresty/patches/LuaJIT-2.1-20210510_01-ffi-arm64-macos-fix-vararg-call-handling.patch (hunk 1 above)
Comment (Water-Melon) [reply build/openresty/patches/LuaJIT-2.1-20210510_01-ffi-arm64-macos-fix-vararg-call-handling.patch:1]: I opened a PR (#13279 ) to remove them.

Comment (AndyZhang0707) [reply build/openresty/patches/LuaJIT-2.1-20210510_01-ffi-arm64-macos-fix-vararg-call-handling.patch:1]: 2.8.4 (also 2.8.3) were built using kbt 4.25.5 and these patch files were in https://github.com/Kong/kong-build-tools/tree/4.25.5/openresty-patches/patches/1.19.9.1 . During the bazel migration we've migrated them into the kong repository, so they're not new patches.

I've diffed this PR's patch files and the kbt 4.25.5 patches and the only differences are hash fix plus one build config change. The build config change is needed by bazel build:

image

File: .requirements (hunk 3 above)
Comment (Water-Melon) [reply .requirements:12]: We downgrade these dependencies' versions to try to keep them consistent with version 2.8.4.
https://github.com/Kong/kong/pull/13282

**Review by locao — APPROVED** [review]

**Review by Hayk-S — APPROVED** [review]

**Pull request merged**